include/oneapi/dpl/internal/dynamic_selection_impl/auto_tune_policy.h

```diff
@@ -217,6 +218,11 @@ class auto_tune_policy
     select(Function&& f, Args&&... args)
     {
         static_assert(sizeof...(KeyArgs) == sizeof...(Args));
+        if constexpr(backend_traits::lazy_report_v<Backend> && backend_traits::enable_profiling_v<Backend>){
+            if(backend_->has_enable_profiling == true){
+                backend_->lazy_report();
+            }
+        }
```

Comment:
Yes, I considered this. However, this would mean adding lazy_report to every new backend. And the usage of lazy_report for now is limited to only sycl::queue resource that have sycl_profiling enabled.

Reply:
You can check existence of `lazy_report` with a compile-time backend trait, that's still OK. I am only asking to hide the other trait, which seems more an implementation detail.

Reply:
I agree that lazy_report is enough. No need for the other enable_profiling trait.

Reply:
Fixed